### PR TITLE
Add support LuaJIT

### PR DESCRIPTION
**Pull request by komainu8**
This pull request for bundling LuaJIT in Groonga.
However, I think this pull request is not enough implementation.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -897,6 +897,32 @@ else
...
+                      [GRN_WITH_LUAJIT=yes],
+                      [GRN_WITH_LUAJIT=no])
+  ],
+  [GRN_WITH_LUAJIT=no])
+  if test "$GRN_WITH_LUAJIT" = "yes"; then
+    AC_DEFINE(GRN_WITH_LUAJIT, [1],
+              [Support LuaJIT.])
+  else
+    if test "x$with_luajit" != "xauto"; then
+      AC_MSG_ERROR("No libluajit found")
```
Comment (kou) [inline configure.ac]: "libluajit" -> "LuaJIT"

File: configure.ac [hunk 2]
```diff
@@ -897,6 +897,32 @@ else
...
 
+# LuaJIT
+AC_ARG_WITH(luajit,
+  [AS_HELP_STRING([--with-luajit],
+    [Support LuaJIT. [default=auto]])],
+  [with_luajit="$withval"],
+  [with_luajit="auto"])
+if test "x$with_luajit" != "xno"; then
+  m4_ifdef([PKG_CHECK_MODULES], [
+    PKG_CHECK_MODULES([LUAJIT],
```
Comment (kou) [inline configure.ac]: We need one more indent.

File: configure.ac [hunk 3]
```diff
@@ -897,6 +897,32 @@ else
...
+    AC_DEFINE(GRN_WITH_LUAJIT, [1],
+              [Support LuaJIT.])
+  else
+    if test "x$with_luajit" != "xauto"; then
+      AC_MSG_ERROR("No libluajit found")
+    fi
+  fi
+else
+  GRN_WITH_LUAJIT=no
+fi
```
Comment (kou) [inline configure.ac]: We need `AC_SUBST(LUAJIT_CFLAGS)` and `AC_SUBST(LUAJIT_LIBS)` to use them in `Makefile.am`.

**Comment by komainu8** [comment]
@kou Thank you for your comments.
I've fixed them.

File: lib/Makefile.am [hunk 4]
```diff
@@ -29,7 +29,8 @@ AM_CXXFLAGS =					\
 
 BUNDLED_LIBRARIES_CFLAGS =			\
 	$(MRUBY_CFLAGS)				\
-	$(ONIGMO_CFLAGS)
+	$(ONIGMO_CFLAGS)                        \
+	$(LUAJIT_CFLAGS)
```
Comment (kou) [inline lib/Makefile.am]: LuaJIT isn't a bundled library.
`libgroonga_la_LIBADD` is better.

Comment (kou) [reply lib/Makefile.am]: Ah, `AM_CFLAGS` for `LUAJIT_CFLAGS`.

**Comment by komainu8** [comment]
@kou Thank you for your comments
I've fixed them and I've added install LuaJIT in travis in 17a4189 .

**Comment by komainu8** [comment]
I found some mistakes.
Therefore, I've fixed them.

File: lib/Makefile.am [hunk 5]
```diff
@@ -58,7 +59,7 @@ libgroonga_la_LIBADD =				\
 	mrb/libgrnmrb.la			\
 	proc/libgrnproc.la			\
 	ts/libgrnts.la				\
-	$(MESSAGE_PACK_LIBS)
+	$(MESSAGE_PACK_LIBS)
```
Comment (kou) [inline lib/Makefile.am]: There is a needless trailing space.

**Comment by komainu8** [comment]
@kou Thank you for your comment.
I've removed a needless trailing space.

**Pull request merged**